mongoengine/queryset/queryset.py

```diff
@@ -1276,10 +1292,10 @@ def _cursor(self):
             # level, not a cursor level. Thus, if read preference is defined,
             # we need to get a cloned collection object using `with_options`
             # first.
-            if self._read_preference is not None:
+            if self._read_preference is not None or self._read_concern is not None:
                 self._cursor_obj = (
                     self._collection
-                        .with_options(read_preference=self._read_preference)
+                        .with_options(read_preference=self._read_preference, read_concern=self._read_concern)
```

Comment:
Just checking: Is it OK to explicitly pass `read_preference=None` if `self._read_preference` is not set? Same with `self._read_concern`?

Reply:
Yes - the defaults are `None` for both - see [docs](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.with_options).